git-hooks/pre-commit

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+from contextlib import contextmanager
+
+ABORT = 1
+
+
+if __name__ == '__main__':
+    try:
+        abort = False
+        abort_messages = []
+        skip_flake8 = False
+        FLAKE8_CFG_FILE = '.flake8'
+
+        # Get base folder of repo
+        loc_proc = subprocess.Popen('git rev-parse --show-toplevel'.split(), stdout=subprocess.PIPE)
+        repo_base_folder = loc_proc.communicate()[0].decode('utf-8').strip()
+        flake8_config_path = os.path.join(repo_base_folder, FLAKE8_CFG_FILE)
+
```

Comment:
Minor, but I think flake8 itself will complain about two blank lines here. Are you running the linter on this code?

Reply:
I'm not running it on this code, but the linter in the workflow will only apply on files with a ".py" extension.

Reply:
Ah, interesting. That makes sense, but given that this _is_ a python file--even if it doesn't have the file extension--I think it's the right thing to run the linter on it before we add it to the repo.